crates/polars-time/src/truncate.rs

```diff
@@ -14,16 +14,53 @@ pub trait PolarsTruncate {
 
 impl PolarsTruncate for DatetimeChunked {
     fn truncate(&self, tz: Option<&Tz>, every: &StringChunked, offset: &str) -> PolarsResult<Self> {
-        let offset = Duration::parse(offset);
+        let offset: Duration = Duration::parse(offset);
+        let time_zone = self.time_zone();
+        // A sqrt(n) cache is not too small, not too large.
+        let mut duration_cache = FastFixedCache::new((every.len() as f64).sqrt() as usize);
```

Comment:
Can we allocate the cache below the fast path. The fast path only accesses the cache (and misses) once.